### ISSUE TITLE
Missing drawing components when combining aplot with geom_fruit

### ISSUE THREAD

**Issue by dinhe878** [YuLab-SMU/aplot#15]
Hi,

I've been struggling to generate a particular composite plot. Nearly all is good except one scale went missing. Hard to explain with text so here are some figures:

Tree figure - 

```
onTip_data <- genus_metadata %>%
  select(c(Diet_specialization, Num_species)) %>%
  rownames_to_column("tip_lable")

g <- ggtree(full_host_genus.tre, right=T, branch.length = 'none') %<+% 
  onTip_data + 
  geom_tippoint(aes(size = Num_species, shape = Diet_specialization)) +
  scale_size(trans="exp", breaks=c(1,2,3)) +
  theme(legend.position = "left")
```

![image](https://user-images.githubusercontent.com/13674319/150346952-a9bde1a1-f252-4c9c-b06a-1c940f6360e5.png)

Then combining with a boxplot using geom_fruit -

```
g1 <- g + geom_fruit(data=all_metadata, 
                     geom=geom_boxplot,
                     aes(x=Shannon, y=genus),
                     pwidth=1, offset=0.2,
                     grid.params=list(alpha=.5),
                     axis.params=list(axis="x", text=c(0,1,2,3,4,5,6), text.size=2, vjust=1))
```

![image](https://user-images.githubusercontent.com/13674319/150347025-a1645498-df2d-4e4b-8974-35e8bf056f3d.png)

Finally combining with a barplot using aplot -

```
p <- plot_composition(genus_r70k_miAbundFamily.ps, 
                      taxonomic.level="Family", 
                      sample.sort=order_name_list)

p1 <- p + scale_fill_brewer("Microbiota (Family)", palette = "Paired") + 
  guides(fill = guide_legend(ncol = 1)) +
  labs(x="", y="Relative abundance") +
  scale_y_percent() + 
  coord_flip() + 
  theme_ipsum(grid="Y") +
  theme(axis.text.x = element_text(angle=35, hjust=1),
        axis.text.y = element_text(angle=0, hjust=1),
        legend.text = element_text(face = "italic"))

p1 %>% insert_left(g1, width=1)
```

![image](https://user-images.githubusercontent.com/13674319/150347300-6bf825d1-4421-48f8-a4e5-71fe2491aea5.png)

As you can see that the scale lines from the "Shannon" boxplot are missing and it returned these warnings:

> Warning messages:
> 1: Removed 9 rows containing missing values (geom_segment). 
> 2: Removed 8 rows containing missing values (geom_segment). 
> 3: Removed 7 rows containing missing values (geom_text). 

I even try to manually add vertical lines to the final figure but no luck there. Advice please!

Many thanks!
/Ding

**Comment by GuangchuangYu** [comment]
@xiangpin

**Comment by xiangpin** [comment]
This issue has been fixed by `ggtreeExtra` (>=1.4.1), which is the [newest released version](https://bioconductor.org/packages/release/bioc/html/ggtreeExtra.html).

**Comment by dinhe878** [comment]
Thanks a lot for the help. However, my issue persists. I've tried the latest version of ggtreeExtra (1.5.2) and it still produce the same plot with missing scale lines of the boxplot and the warning messages (one extra):

> Warning messages:
> 1: Removed 9 rows containing missing values (geom_segment). 
> 2: Removed 8 rows containing missing values (geom_segment). 
> 3: Removed 7 rows containing missing values (geom_text). 
> 4: Removed 1 rows containing missing values (geom_text). 

I ended up instead, using cowplot package to "hard" align the figures.

`ggdraw() +
  draw_plot(g1, x=0, y=.07, width=.3, height=.915) +
  draw_plot(p1, x=.25, y=0, width=.7, height=1)`

![GAGA_genus_TAXAbarplot_shannon](https://user-images.githubusercontent.com/13674319/151949303-da9bfc9e-194f-4809-bbda-c1aaf0e772e9.png)

But I still think that it'd good to know what went wrong with my previous approach, if possible.

Many thanks again for making these great tools!

Best, Ding
